content/usgs-packages/sbtools_Modify.Rmd

```diff
@@ -0,0 +1,302 @@
+
+To update an existing item, use the `item_update` function. It requires two arguments: the `sbitem` and a list of the metadata key-values of what to change. Change the title of the item created earlier and saved as `test_item` in R to "sbtools example data". Then, look at the item title to verify.
+
+```{r sbtools-update}
+# before
+test_item$title
+
+# update
+updated_item <- item_update(test_item, info=list(title="sbtools example data"))
+
+# after
+updated_item$title
+```
+
+You can also update multiple items at once using the plural of the function, `items_update`. Try adding a contact as metadata to each of the items under your user account. The `info` argument needs to be a list the same length as the number of sbitems you are updating. Each of these lists is a list with metadata key-value pairs. The contact metadata has a strange syntax where you need two additional nested lists. 
+
+```{r sbtools-update-mult}
+all_items <- item_list_children(user_id())
+update_contacts <- items_update(all_items, 
+             info = 
+               list(rep(list(contacts = list(list(name="Julius Caesar"))), 2)))
+
+# verify that the contacts were added to all items
+sapply(all_items, item_get_fields, fields="contacts")
+```
+
+If you want to update an item but it might not exist yet, use the `upsert` functions. Currently, this function is not working as you might expect. There are two `sbtools` issues that will address this: [issue 239](https://github.com/USGS-R/sbtools/issues/239) and [issue 240](https://github.com/USGS-R/sbtools/issues/240).
+
+*EXAMPLES NOT CURRENTLY WORKING*
+
+```{r sbtools-upsert, eval=FALSE}
+# creates a new item even though one already exists
+item_upsert(title="books.json") 
+
+# created another item with the name "books.json" with this message:
+#   "title is NULL - re-using title from input SB item"
+item_upsert(test_item, title=NULL, info=list(title = "sbtools stuff"))
+
+# created a new item named "sbtools stuff" under the 
+# books.json item
+item_upsert(test_item, title="sbtools stuff")
+```
+
+Another way to edit or update SB items is to add alternative identifiers, e.g. digital object identifiers, IPDS codes, etc. They can be useful to identify your items in searches because they have a user-defined type, scheme, and key. For examples of identifiers, see the "Additional Information | Identifiers" section of [Differential Heating](https://www.sciencebase.gov/catalog/item/580587a2e4b0824b2d1c1f23) (two sets of identifiers, one for a DOI and one for an IPDS, each of which has type/scheme/key) and [nwis_01645704](https://www.sciencebase.gov/catalog/item/556f2055e4b0d9246a9fc9f7) (one set of identifiers, created by us in our mda_streams scheme to help us search for items). Here we will show how to add alternative identifiers. See ?`query_item_identifier` if you want to know more about querying by alternative identifiers.
```

Comment:
:+1: consolidation

I'll still add detail about the mda_streams bit here and then consolidate for #272